docs/application/native/guides/device/battery-monitor.md

```diff
@@ -0,0 +1,230 @@
+   The `battery_monitor_resource_id_e`, `battery_monitor_duration_type_e` enums can be used to provide information related to specific resource and particular duration type respectively, as shown in the following code:
+
+   ```
+   #include <battery_monitor.h>
+
+   void
+   func(void)
+   {
+	char* app_id = "org.tizen.samsung";
+
+	int error_code = BATTERY_MONITOR_ERROR_NONE;
+	int battery_usage = -1;
+	battery_monitor_resource_id_e resource_id = BATTERY_MONITOR_RESOURCE_ID_DISPLAY;
+	battery_monitor_duration_type_e duration_val = BATTERY_MONITOR_DURATION_TYPE_1DAY;
+
+	error_code = battery_monitor_get_usage_by_app_id_for_resource_id(app_id, resource_id, duration_val, &battery_usage);
+
+	if (error_code == BATTERY_MONITOR_ERROR_NONE)
+		printf("The Battery Usage for appid [%s], for resource display is [%d]", app_id, battery_usage);
+	else
+		printf("Error Occurred [%d]", error_code);
+
+	resource_id = BATTERY_MONITOR_RESOURCE_ID_WIFI;
+	duration_val = BATTERY_MONITOR_DURATION_TYPE_1WEEK;
+
+	error_code = battery_monitor_get_usage_by_app_id_for_resource_id(app_id, resource_id, duration_val, &battery_usage);
+
+	if (error_code == BATTERY_MONITOR_ERROR_NONE)
+		printf("The Battery Usage for appid [%s], for resource wifi is [%d]", app_id, battery_usage);
+
+	else
+		printf("Error Occurred [%d]", error_code);
+
+	free(app_id);
+
+	return;
+   }
+   ```
+
+2. Fetching total battery usage information of an application ID by combining all the resources over certain a certain duration of time.
+   The `battery_monitor_duration_type_e` enum can be used to choose a particular duration type, see the following code:
+
+   ```
+   #include <battery_monitor.h>
+
+   void
+   func(void)
+   {
+	char* app_id = "org.tizen.samsung";
+
+	int error_code = BATTERY_MONITOR_ERROR_NONE;
+	int battery_usage = -1;
+	battery_monitor_duration_type_e duration_val = BATTERY_MONITOR_DURATION_TYPE_1DAY;
+
+	error_code = battery_monitor_get_total_usage_by_app_id(app_id, duration_val, &battery_usage);
+	if (error_code == BATTERY_MONITOR_ERROR_NONE)
+		printf("The total battery usage for appid [%s] over last 24 hrs is [%d]", app_id, battery_usage);
+	else
+		printf("Error Occurred [%d]", error_code);
+
+	duration_val = BATTERY_MONITOR_DURATION_TYPE_1WEEK;
+	error_code = battery_monitor_get_total_usage_by_app_id(app_id, duration_val, &battery_usage);
+	if (error_code == BATTERY_MONITOR_ERROR_NONE)
+		printf("The total battery usage for appid [%s] over last 7 days is [%d]", app_id, battery_usage);
+	else
+		printf("Error Occurred [%d]", error_code);
+
+	return;
+   }
+   ```
+3. Fetching battery usage values for all the resources used by an application ID for a certain duration of time.
+   The `battery_monitor_resource_id_e`, `battery_monitor_duration_type_e` enums can be used to provide information related to specific resource and particular duration type respectively, as shown in the following code:
+
+   ```
```

Comment:
Suggestion for this sample code.
Many part are duplicated with above sample. How about just showing the specific part only for this section?

as like 

   ```
	battery_monitor_h data_handle = NULL;

	/*First use this API to fetch the usage for all the available resources in the data handle*/
	error_code = battery_monitor_get_usage_by_app_id_for_all_resource_id(app_id, duration_val, &data_handle);

	if (error_code == BATTERY_MONITOR_ERROR_NONE)
		printf("Data Handle information received");
	else
		printf("Error Occurred [%d]", error_code);

	/*Now retrieving the information one by one from the data_handle*/
	battery_monitor_resource_id_e resource_id = BATTERY_MONITOR_RESOURCE_ID_DISPLAY;

	error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage);
	if (error_code == BATTERY_MONITOR_ERROR_NONE)
		printf("Battery Usage by display is [%d]", battery_usage);

	resource_id = BATTERY_MONITOR_RESOURCE_ID_WIFI;

	error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage)

	if (error_code == BATTERY_MONITOR_ERROR_NONE)
		printf("Battery Usage by display is [%d]", battery_usage);

	/* Similarly for other resources...
	.
	.
	.
	*/

	error_code = battery_monitor_destroy(data_handle);
	if (error_code != BATTERY_MONITOR_ERROR_NONE)
		printf("Error Occurred [%d]", error_code);
   ```